Bump github/codeql-action from 3.28.3 to 3.30.3 (#395)
rsksmart/devportal: rif/rns/guide/value-proposition/wallets.md

---
layout: rsk

> title: "Value proposition RNS for Wallets"

Reviewer: Sure. What do you propose?

Another idea:
"RNS | Full Integration Guide & Value proposition"

> title: "Value proposition RNS for Wallets"

Reviewer: great!

description: "Phase 1: Problems to solve, UX proposal, Send BTC example, Feature summary, RNS Integration guidelines. Phase 2:  Problems to solve, Feature summary"
tags: rns, guide, wallets, value
---

## Phase 1

> **Improve UX to improve user adoption**

Reviewer: perfect

*   Simplify the way users send and receive crypto-assets (in RSK and other blockchains)
*   Avoid losing funds due to human errors and hacking/phishing.

### Problems to solve

1. **Difficult or impossible to remember crypto wallet addresses**
   _→ bad UX for end users, diminishes non-tech user adoption._
2. **Error-prone when writing traditional crypto wallet addresses**
   _→ can derive to lost funds when transferring crypto assets._
3. **Hacking/phising-prone when copy-pasting traditional crypto wallet addresses**

>    _→ can derive to lost funds when transferring crypto assets._

Reviewer: Both? Do you have a proposal to change them? Do you need more input from my side?

>    _→ can derive to lost funds when transferring crypto assets._

Reviewer: perfect

4. **Difficult to add contacts to crypto wallets** **and impossible to dynamically update contact’s addresses**
  _→ bad UX for end users._

### UX Proposal

#### Full UX Flow in Figma

[https://www.figma.com/file/U1lTWDKjzIi8JvEl77FEYy/RNS-for-rWallet](https://www.figma.com/file/sRGQFRnFTJXcVpxSGBwZIT/RNS-for-Wallet-Integration?node-id=0%3A1&viewport=421%2C-606%2C0.07086607068777084)

**_Steps:_**

> 1. _Click the above [link](https://www.figma.com/file/sRGQFRnFTJXcVpxSGBwZIT/RNS-for-Wallet-Integration?node-id=0%3A1&viewport=421%2C-606%2C0.07086607068777084) in desktop computer._

Reviewer: perfect

   ![RNS for Wallets - Integration](/rif/rns/guide/images/rns-for-wallets-integration.png)

### Quick example: Send BTC

![RNS for Wallets - Walkthrough 01](/rif/rns/guide/images/rns-for-wallets-walkthrough-01.png)

![RNS for Wallets - Walkthrough 02](/rif/rns/guide/images/rns-for-wallets-walkthrough-02.png)

![RNS for Wallets - Walkthrough 03](/rif/rns/guide/images/rns-for-wallets-walkthrough-03.png)

![RNS for Wallets - Walkthrough 04](/rif/rns/guide/images/rns-for-wallets-walkthrough-04.png)

![RNS for Wallets - Walkthrough 05](/rif/rns/guide/images/rns-for-wallets-walkthrough-05.png)

This will send the funds to the app user’s Bitcoin address linked to `alice.yourwallet.rsk`.

### Feature summary

1. **YourWallet Nickname**
    *   _Free_ RNS subdomain when installing app**
        1. Domain template: `xxxxx.yourwallet.rsk`
        2. `xxxxx` is the nickname
    *   **It is linked to all the wallet addresses created when installing the app.**
    *   It can be presented to end users as a “[YourWallet]’s **nickname**” to improve non-tech user adoption.
    *   Users won’t have to understand a new concept such as RNS nor smart contracts nor fees nor gas.

>     *   Behind the scenes is simply a RNS subdomain of the yourwallet.rsk domain

Reviewer: ok

2. **Send crypto-asset using human-readable addresses** (_YourWallet_ nicknames and any other RNS domain) while also maintaining traditional wallet addresses.
3. **Contact book**
4. **Easy _YourWallet_ nickname sharing**

### RNS Integration Guidelines

[https://developers.rsk.co/rif/rns/integrate/](https://developers.rsk.co/rif/rns/integrate/)

### RNS Documentation

#### rns-js library

[https://www.npmjs.com/package/@rsksmart/rns](https://www.npmjs.com/package/@rsksmart/rns)

#### RNS in RSK Developer portal

[https://developers.rsk.co/rif/rns/](https://developers.rsk.co/rif/rns/)

#### Integrate Your Wallet with RNS

[https://developers.rsk.co/rif/rns/integrate/integrate-wallet/](https://developers.rsk.co/rif/rns/integrate/integrate-wallet/)

#### Integrate Your dApp with RNS

[https://developers.rsk.co/rif/rns/integrate/integrate-dapp/](https://developers.rsk.co/rif/rns/integrate/integrate-dapp/)

## Phase 2 (optional)

**Empower YourWallet users with the full RNS experience**

*   Let Your Wallet users have full control of their RNS domains.
*   Let users switch their main RNS domain in Your Wallet.
*   Let users register new RNS domains.

> *   Let users admin their RNS domains from Your Wallet.

Reviewer 2: ```suggestion
*   Let users be the admin of their RNS domains from Your Wallet.
```

Reviewer: My original intention was to use the "admin" word as a verb.
I want to emphasize that the users will be able to do actions, manage/admin their own domains.

Maybe this is better?: "Let users manage their own RNS domains from Your Wallet"

Reviewer 3: ```suggestion
*   Let users manage their own RNS domains from Your Wallet
```

Agreed, "manage" is much clearer in this context. I think you would have to use the non-contracted form "administrate" in order to convey that it is a verb as clearly. Anyhow, going with your final suggestion.

> ### Problems to solve
> 

Reviewer 3: ```suggestion
### Problems to solve

Upon completing **phase 1**, your wallet would have the minimum viable functionality needed for an RNS integration. However, there are some additional features, that we suggest below, that would further improve the user experience.

```

Reviewer: perfect

1. **Not possible to update the address resolution for my domain.**

>     _→ more steps to do it if having to go to the [RNS Manager](https://rns.rifos.org)_

Reviewer 2: needs to be rephrased

Reviewer 3: agreed, I have suggested a new paragraph for this section (above) that helps to explain the intent a bit better.

```suggestion
    _→ user needs to perform additional steps by using the [RNS Manager](https://rns.rifos.org)_
```

Reviewer: perfect

2. **Not possible to change my main YourWallet subdomain (nickname)**
    _→ less flexibility for end users and less decentralization_
3. **Not possible to control, register, use and admin different RNS domains from YourWallet**
    _→ RNS frequent users will have less flexibility to fulfill their needs_

> 
> 
> ### Feature Summary

Reviewer 3: ```suggestion

### Feature Summary
```
